### PR TITLE
[PB-687] feat/calculate-bandwidth

### PR DESCRIPTION
**Pull request by miguelsw**
- Add upload and download bandwith to send in track options.

### CODE REVIEW

File: src/app/drive/services/download.service/downloadFile.ts [hunk 1]
```diff
@@ -155,7 +161,7 @@ export default async function downloadFile(
     if (connectionLost) throw new ConnectionLostError();
     else throw err;
   }
-
+  console.log('2');
```
Comment (sg-gs) [inline src/app/drive/services/download.service/downloadFile.ts]: Remove this `console.log`

File: src/app/drive/services/network.service/index.ts [hunk 2]
```diff
@@ -223,4 +223,33 @@ export function getEnvironmentConfig(isTeam?: boolean): EnvironmentConfig {
   };
 }
 
+export class Band {
+  private _startTime: number;
```
Comment (sg-gs) [inline src/app/drive/services/network.service/index.ts]: We are already using `private` in TypeScript, the `_` convention is not required

File: src/app/drive/services/network.service/index.ts [hunk 3]
```diff
@@ -223,4 +223,33 @@ export function getEnvironmentConfig(isTeam?: boolean): EnvironmentConfig {
...
+  setEndTime() {
+    this._endTime = new Date().getTime();
+  }
+  setSize(size: number) {
+    this._size = size;
+  }
+  getSize() {
+    return this._size;
+  }
+  getStartTime() {
```
Comment (sg-gs) [inline src/app/drive/services/network.service/index.ts]: use getters and setters

**Review by sg-gs — CHANGES_REQUESTED** [review]

File: src/app/drive/services/network.service/index.ts [hunk 4]
```diff
@@ -223,4 +223,23 @@ export function getEnvironmentConfig(isTeam?: boolean): EnvironmentConfig {
...
+  private endTime: number = 0;
+  private size: number = 0;
+
+  constructor() {
+    this.startTime = new Date().getTime();
+  }
+  addEndTime(): void {
+    this.endTime = new Date().getTime();
+  }
+  set sise(size: number) {
```
Comment (CandelR) [inline src/app/drive/services/network.service/index.ts]: typo: `sise` -> `size`

File: src/app/drive/services/folder.service.ts [hunk 5]
```diff
@@ -339,6 +340,7 @@ async function downloadFolderAsZip(
           };
           analyticsService.trackFileDownloadStarted(trackingDownloadProperties);
           console.log({ trackingDownloadProperties });
```
Comment (CandelR) [inline src/app/drive/services/folder.service.ts]: Remove this console.log

**Review by CandelR — APPROVED** [review]

File: src/app/drive/services/network.service/index.ts [hunk 6]
```diff
@@ -243,4 +243,23 @@ export function getEnvironmentConfig(isTeam?: boolean): EnvironmentConfig {
...
+  }
+  addEndTime(): void {
+    this.endTime = new Date().getTime();
+  }
+  set setSize(size: number) {
+    this.size = size;
+  }
+  getBandwith(): number {
+    const totalTimeSeconds = (this.endTime - this.startTime) / 1000; // seconds
+    const totalMB = this.size / 1024 / 1024; //MB
```
Comment (SimonGodefroid) [inline src/app/drive/services/network.service/index.ts:262]:
```suggestion
    const totalMB = this.size / 1024 / 1024; // MB
```

**Comment by sg-gs** [comment]
Not tracking this anymore, closing @internxt/web

**Pull request closed without merging**